### DISCUSSION TITLE
EV3 - Mindsensor Library for Pybricks

### DISCUSSION THREAD

**Discussion by alpha71** [pybricks/support#337] [General]
Hello,
these days i was talking with Mindsensor (http://www.mindsensors.com) regarding several EV3 sensors for which i asked sopport to implement them in the PyBricks library. actually they're preparing (still in working progress!) and i'd like to share with you what i got, of course the library is still in progress so better to check in their repository to any further updating!
Here below the list of sensors supported until now.
One question is: **WHERE SHOULD LOCATE THE FILE _mindsensorPYB.py_**?

```python
from  mindsensorsPYB import PPS58
from  mindsensorsPYB import EV3Matrix
from  mindsensorsPYB import DIST_ToF
from  mindsensorsPYB import EV3RFid
from  mindsensorsPYB import LINELEADER
from  mindsensorsPYB import TFTPACK
from  mindsensorsPYB import ABSIMU
from  mindsensorsPYB import IRThermometer
from  mindsensorsPYB import VOLT
from  mindsensorsPYB import PFMATE`
```

[Mindsensor.zip](https://github.com/pybricks/support/files/6472911/Mindsensor.zip)

**Comment by dlech** [comment]
You can place the file in `~/.pybricks-micropython/lib/` (aka `/home/robot/.pybricks-micropython/lib/`) to make it available for all of your programs.

You can also install `pybricks-mpy-cross` and then use it to precompile it to a .mpy file and place that file in the directory instead to improve startup time a tiny bit.

    sudo apt update && sudo apt install pybricks-mpy-cross
    mkdir -p ~/.pybricks-micropython/lib/
    pybricks-mpy-cross mindsensorsPYB.py
    mv mindsensorsPYB.mpy ~/.pybricks-micropython/lib/

**Reply by alpha71** [reply]
Hello David,
i'm sorry but i'm a bit lost ... then please forgive my follow questions:
1. how to install it? from which editor?
2. how to place in the root directory, coz i cannot see it from the EV3DEV DEVICE BROWSER in the Visual Studio!


![immagine](https://user-images.githubusercontent.com/37776195/118187797-be163d80-b43f-11eb-99ba-ae3d3e929e4a.png)

**Reply by alpha71** [reply]
At the end i use the SSH Terminal, but i encored in some problem, I send you a screenshot after lunch the first comand line:
`sudo apt update && sudo apt install pybricks-mpy-cross`
It seems has some error to access to the archive of _ev3dev.org_
And still i don't know how to copy the library file into the home/robot/...

![immagine](https://user-images.githubusercontent.com/37776195/118190747-d7b98400-b443-11eb-9f48-2dd101869451.png)

**Reply by alpha71** [reply]
i'm really very poor in Linux!!!!

**Reply by dlech** [reply]
The problem with `apt` is that the EV3 is not connected to the Internet. But this step is not necessary.

**Reply by dlech** [reply]
You can use WinSCP or Filezilla to copy files to the EV3. Or as Laurens suggested, you can just include a copy of the file in each of your project folders.

**Comment by laurensvalk** [comment]
How about including the files you need in your project folder?

When you run a program, the whole folder gets uploaded to the brick.

**Reply by alpha71** [reply]
I tried to copy the file in the working folder and also in the `~/.pybricks-micropython/lib/` but still cannot works!
plus the installation of `pybricks-mpy-cross` has some trouble. Actaully i downloaded for your Github (trhe version 5), it a file `*.tgz` but i don't know how to use it!


![immagine](https://user-images.githubusercontent.com/37776195/118196061-a5605480-b44c-11eb-8ce8-ac5b08d99aae.png)

![immagine](https://user-images.githubusercontent.com/37776195/118196155-ccb72180-b44c-11eb-9a3a-fcd05fb3df6a.png)

**Reply by dlech** [reply]
It looks like the spelling is wrong. The file is `mindsensorPYB.py` but you are importing `mindsensorsPYB`. The import has to match the name of the file.

Don't worry about `pybricks-mpy-cross`.

**Reply by alpha71** [reply]
Hello, ok finally I figured out... yes, it was a spelling error :( (how come?!? no words to myself!!!). BTW now it seems works. 
I'm sorry occupied your time due to my lack of Linux knowledge!!! :(

![immagine](https://user-images.githubusercontent.com/37776195/118294098-ab9d1200-b4da-11eb-8133-643088d41cb8.png)
